theseus/labs/lie/lie_tensor.py

```diff
@@ -143,6 +166,20 @@ def _no_jop(self, input0: TensorType) -> _JFnReturnType:
     jleft_act = _no_jop
     jleft_project = _no_jop
 
+    def retract(self, delta: TensorType) -> "LieTensor":
+        assert isinstance(delta, torch.Tensor)
+        return self.compose(self.exp(delta))
+
+    def __add__(self, other: TensorType) -> "LieTensor":
+        if not isinstance(other, TangentTensor):
+            raise RuntimeError(
+                "Operator + is only supported for tensors of ltype=tgt. "
+                "If you intend to add the raw tensor data, please use group._t. "
```

Comment:
Rather than using `torch._t`, we should provide some method that return the group's data as a raw torch tensor (e.g., `g.mat()` or maybe `g.as_torch(fmt="mat")`.

Reply:
`as_tensor` with an optional format passed, sounds good. Could you use cast here?

Reply:
`mat` seems to have different meanings for Lie groups, which depends on the representation. For example, if we use quaternions to represent SO3, `g.mat()` should be a `3x3` matrix rather than a `4x1` vector.

Reply:
`mat` means **always** return a matrix independently of what the internal representation is. My vision is the actual representation is hidden from the user, and they can get access to tensor data with something like `mat()`, `quat()`, etc. BTW, I just realized that this also means that we don't need `to_euclidean()`, methods such as `mat()` and `quat()` would cover this functionality.

Reply:
Sounds good that direct access to internal storage is hidden and mat/quat return appropriate sized tensors. But for us internally are we storing things as mat or quat, and is this a setting exposed to user in case they want to switch?